crates/common/types/transaction.rs

```diff
@@ -54,7 +54,7 @@ pub struct EIP2930Transaction {
     pub signature_s: U256,
 }
 
-#[derive(Clone, Debug, PartialEq, Eq, Default)]
+#[derive(Clone, Debug, PartialEq, Eq, Default, Deserialize)]
```

Comment:
This struct has a custom `Serialize` implementation, directly deriving `Deserialize` here will make these traits incompatible and will lead to potential bugs down the line